modules/account-lib/src/coin/stacks/utils.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  AddressHashMode,
+  StacksTransaction,
+  TransactionVersion,
+  addressFromVersionHash,
+  addressHashModeToVersion,
+  addressToString,
+  validateStacksAddress,
+} from '@stacks/transactions';
+import BigNumber from 'bignumber.js';
+
+/**
+ * Encodes a buffer as a `0x` prefixed lower-case hex string.
+ *
+ * @param {Buffer} buff - a buffer with a hexadecimal string
+ * @returns {string} - the hexadecimal string prefixed with "0x"
+ */
+export function bufferToHexPrefixString(buff: Buffer): string {
+  return '0x' + buff.toString('hex');
+}
+
+/**
+ * @param hex
+ */
+export function removeHexPrefix(hex: string): string {
+  if (hex.startsWith('0x')) return hex.slice(2);
+  else return hex;
+}
+
+/**
+ * @param publicKeyHash
+ * @param hashMode
+ * @param transactionVersion
+ */
+function getAddressFromPublicKeyHash(
+  publicKeyHash: Buffer,
+  hashMode: AddressHashMode,
+  transactionVersion: TransactionVersion,
+): string {
+  const addrVer = addressHashModeToVersion(hashMode, transactionVersion);
+  if (publicKeyHash.length !== 20) {
+    throw new Error('expected 20-byte pubkeyhash');
+  }
+  const addr = addressFromVersionHash(addrVer, publicKeyHash.toString('hex'));
+  const addrString = addressToString(addr);
+  return addrString;
+}
+
+/**
+ * @param tx
+ */
+export function getTxSenderAddress(tx: StacksTransaction): string {
+  if (tx.auth.spendingCondition !== null && tx.auth.spendingCondition !== undefined) {
```

Comment:
https://github.com/stacksgov/sips/blob/main/sips/sip-005/sip-005-blocks-and-transactions.md#transaction-authorization

want to make sure we cover for sponsored transactions as well - do we clean/verify for both spending conditions?